activerecord/lib/active_record/relation.rb

```diff
@@ -1023,7 +1023,11 @@ def _scoping(scope, registry, all_queries = false)
       def _substitute_values(values)
         values.map do |name, value|
           attr = table[name]
-          unless Arel.arel_node?(value)
+          if Arel.arel_node?(value)
+            if value.is_a?(Arel::Nodes::SqlLiteral) && value.match?(/\A[^(].*\W/)
```

Comment:
Why this regexp? Is it to avoid double parantheses? I don't see a big problem with it, and even then it's not checking the literal also ends with a `)`.

Reply:
Yes, that is to avoid double parentheses. It does not check for ending by `)`, because it is assumed that if we have an opening, then we also have a closing.

So do you suggest that it is not worth it and can be just removed?

Reply:
That's my suggestion yes. It makes the generated SQL sligthly less nice, but the code much simpler

Reply:
Yeah, probably if someone already passed an arel literal, then the query piece is already complex and an extra `()` won't make too much difference.

So, updated with the suggestion, thank you!